File: pod/15sep.py

def main():
    n = int(input())
    if(n == 1):
        print("impossible")
        return(0)
    num_lines = 2*n-1
    num_stars = n
    num_spaces = 1
    for _ in range(n):
        for i in range(num_stars):
            print('*',end='')
        for i in range(num_spaces):
            print(' ',end='')
        for i in range(num_stars):
            print('*',end='')
        print()
        num_spaces+= 2
        num_stars-=1
    num_stars+=1
    num_spaces-=2
    for _ in range(n-1):
        num_spaces -= 2
        num_stars+=1
        for i in range(num_stars):
            print('*',end='')
        for i in range(num_spaces):
            print(' ',end='')
        for i in range(num_stars):
            print('*',end='')
        print()
    return(0)


main()

File: pod/15sep.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n;
    cin>>n;
    if(n == 1){
        cout<<"impossible"<<endl;
        return 0;
    }
    int num_lines = 2*n-1;
    int num_stars = n;
    int num_spaces = 1;
    for(int _ = 0; _ < n; _++){
        for(int i = 0; i < num_stars; i ++){
            cout<<'*';
        }
        for(int i = 0; i < num_spaces; i++){
            cout<<' ';
        }
        for(int i = 0; i < num_stars; i++){
            cout<<'*';
        }
        cout<<endl;
        num_spaces += 2;
        num_stars--;
    }
    num_stars++;
    num_spaces-=2;
    for(int _ = 0; _ < n-1; _++){
        num_spaces -= 2;
        num_stars++;
        for(int i = 0; i < num_stars; i ++){
            cout<<'*';
        }
        for(int i = 0; i < num_spaces; i++){
            cout<<' ';
        }
        for(int i = 0; i < num_stars; i++){
            cout<<'*';
        }
        cout<<endl;
    }
    return 0;
}